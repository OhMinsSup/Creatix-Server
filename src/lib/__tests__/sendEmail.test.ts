import faker from 'faker';
import { sendMail } from '../sendEmail';
import { createAuthEmail } from '../createAuthEmail';

// https://www.moakt.com/ko 을 통해서 가짜 이메일 가져온다
const email = 'zliglmq8t@moakt.co';
const uuid = faker.random.uuid();
const emailTemplate = createAuthEmail(false, uuid);

describe('Send Email Function', () => {
  test('메일 보내기', async () => {
    const transporter = await sendMail({
      to: email,
      ...emailTemplate,
      from: 'verify@creatix.io'
    });
    expect(transporter.envelope).toEqual({
      from: 'verify@creatix.io',
      to: [email]
    });
  });
});
