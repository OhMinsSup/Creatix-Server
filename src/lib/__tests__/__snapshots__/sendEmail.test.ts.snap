// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send Email Function 메일 보내기 1`] = `
Object {
  "accepted": Array [
    "tx3eflk5u@tmpmail.net",
  ],
  "envelope": Object {
    "from": "verify@creatix.io",
    "to": Array [
      "tx3eflk5u@tmpmail.net",
    ],
  },
  "envelopeTime": 768,
  "messageId": "<63952cdd-3bea-588f-20cf-fa6a79decfbb@creatix.io>",
  "messageSize": 2457,
  "messageTime": 815,
  "rejected": Array [],
  "response": "250 2.0.0 OK  1554467734 25sm27022576pfo.96 - gsmtp",
}
`;
